app/src/main/java/com/keylesspalace/tusky/adapter/SearchResultsAdapter.java

```diff
@@ -58,44 +83,83 @@ public SearchResultsAdapter(LinkListener listener) {
-            holder.setupWithAccount(accountList.get(position));
-            holder.setupLinkListener(linkListener);
-        } else {
-            HashtagViewHolder holder = (HashtagViewHolder) viewHolder;
-            int index = position - accountList.size();
-            holder.setup(hashtagList.get(index), linkListener);
+    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder viewHolder, int position) {
+            if (position >= accountList.size()) {
+                if(position >= accountList.size() + concreteStatusList.size()) {
+                    HashtagViewHolder holder = (HashtagViewHolder) viewHolder;
+                    int index = position - accountList.size() - statusList.size();
+                    holder.setup(hashtagList.get(index), linkListener);
+                } else {
+                    StatusViewHolder holder = (StatusViewHolder) viewHolder;
+                    int index = position - accountList.size();
+                    holder.setupWithStatus(concreteStatusList.get(index), statusListener, mediaPreviewsEnabled);
+                }
+            } else {
+                AccountViewHolder holder = (AccountViewHolder) viewHolder;
+                holder.setupWithAccount(accountList.get(position));
+                holder.setupLinkListener(linkListener);
+            }
         }
-    }
 
     @Override
     public int getItemCount() {
-        return accountList.size() + hashtagList.size();
+        return accountList.size() + hashtagList.size() + concreteStatusList.size();
     }
 
     @Override
     public int getItemViewType(int position) {
         if (position >= accountList.size()) {
-            return VIEW_TYPE_HASHTAG;
+            if(position >= accountList.size() + concreteStatusList.size()) {
+                return VIEW_TYPE_HASHTAG;
+            } else {
+                return VIEW_TYPE_STATUS;
+            }
         } else {
             return VIEW_TYPE_ACCOUNT;
         }
     }
 
+    public @Nullable Status getStatusAtPosition(int position) {
+        return statusList.get(position - accountList.size());
+    }
+
+    public @Nullable StatusViewData.Concrete getConcreteStatusAtPosition(int position) {
+        return concreteStatusList.get(position - accountList.size());
+    }
+
+    public void updateStatusAtPosition(StatusViewData.Concrete status, int position) {
+        concreteStatusList.set(position - accountList.size(), status);
+    }
+
+    public void removeStatusAtPosition(int position) {
+        concreteStatusList.remove(position - accountList.size());
+        notifyItemRemoved(position);
+    }
+
     public void updateSearchResults(SearchResults results) {
         if (results != null) {
-            accountList.addAll(results.getAccounts());
-            hashtagList.addAll(results.getHashtags());
+            accountList = results.getAccounts();
+            statusList = results.getStatuses();
+            for(Status status: results.getStatuses()) {
```

Comment:
On previous lines you replace results but here you're appending to the `concreteStatusList` all the time. Is it okay? I cannot test because search for posts doesn't work for me in Mastodon, I'm not sure why.